src/Microsoft.Performance.SDK.Runtime/Discovery/AssemblyExtensionDiscovery.cs

```diff
@@ -452,22 +409,65 @@ public bool ProcessAssemblies(IEnumerable<string> directoryPaths, out ErrorInfo
             return true;
         }
 
+        private static HashSet<string> GetExclusionValues(string targetDirectory, ILogger logger)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(targetDirectory), $"{nameof(targetDirectory)} is null.");
+            Debug.Assert(Directory.Exists(targetDirectory), $"{nameof(targetDirectory)} is invalid.");
+
+            string configurationFile = Directory
```

Comment:
Does this really need to call `GetFiles`? I think a `Path.Combine` followed by a `File.Exists` would suffice